graphdatascience/query_runner/protocol/write_protocols.py

```diff
@@ -94,3 +100,35 @@ def run_write_back(
             False,
             False,
         )
+
+
+class RemoteWriteBackV3(WriteProtocol):
+    def write_back_params(
+        self,
+        graph_name: str,
+        job_id: str,
+        config: Dict[str, Any],
+        arrow_config: Dict[str, Any],
+        database: Optional[str] = None,
+    ) -> CallParameters:
+        return RemoteWriteBackV2().write_back_params(graph_name, job_id, config, arrow_config, database)
+
+    def run_write_back(
+        self, query_runner: QueryRunner, parameters: CallParameters, yields: Optional[List[str]]
+    ) -> DataFrame:
+        def is_not_completed(result: DataFrame) -> bool:
+            result = write_fn().squeeze()
+            return result["status"] != Status.COMPLETED.name
```

Comment:
could extract this into `status.py` as it seems a very generic function :)

Reply:
Let's do that when we have more versions. Unfortunately someone didn't align with the existing return status of the write-back, so that the project procedure will return `DONE` instead, so we at least cannot share everything between write and project